actions/user_actions.jsx

```diff
@@ -143,8 +143,8 @@ export function loadNewDMIfNeeded(channelId) {
         const state = doGetState();
         const currentUserId = Selectors.getCurrentUserId(state);
 
-        function checkPreference() {
-            const userId = Utils.getUserIdFromChannelName(currentUserId);
+        function checkPreference(channel) {
+            const userId = Utils.getUserIdFromChannelName(channel);
```

Comment:
I stumbled over `getUserIdFromChannelName` a few times -- it actually accepts a `channel` and not a channel name, and initially tried to refactor this, but realized there's code duplication here with Redux and opted just to keep this PR simpler.